src/v/cloud_topics/reconciler/reconciler.cc

```diff
@@ -102,6 +106,52 @@ void reconciler::detach(const model::ntp& ntp) {
     }
 }
 
+uint64_t reconciler::compute_max_offset_lag() const {
+    uint64_t max_lag = 0;
+    for (const auto& [_, src] : _sources) {
```

Comment:
This loop could potentially trigger reactor stall. One possible approach here is to `break` if `need_preempt` returns `true` and print log message that says that the estimate is incorrect. Another approach is to sample sources and estimate the bias (not sure it's possible given that distribution is not known in advance). It's better to give inaccurate estimate than stall the reactor.